### ISSUE TITLE
Why vgg19 updates?

### ISSUE THREAD

**Issue by AlbertZhangHIT** [twtygqyy/pytorch-SRResNet#11]
I'm confused whether we need to update the vgg network. The [example](https://github.com/pytorch/examples/tree/master/fast_neural_style) in pytorch tutorial keeps the parameters in vgg16 untouched. But as far as I understand, your implementation updates the parameters. Did I miss something important?

**Comment by twtygqyy** [comment]
Hi @AlbertZhangHIT I didn't update the params in VGG network, just apply the loss for back propagating the sr network.

**Comment by AlbertZhangHIT** [comment]
God my silly question. The parameters should only be updated by optimizers. So the `content_loss` will be backpropagated through VGG, and then the Generator since one candidate of the content loss criterion, the `content_input`, comes from the `input`(which requires gradient), do I get the point? Thanks for your reply.

Edit: Your implementation is different from others' that I viewed where the total loss is usually in a manner of `total_loss = loss1 + loss2 + ...` and then backpropagated. It was a little difficult for me to understand your implementation at the first glimpse because of my poor knowledge about PyTorch.  But your implementation is quite great. Thanks again!
